### PR TITLE
drivers: modem: ublox-sara-r4:  fixes for rssi, udp, sendmsg

### PR DESCRIPTION
**Pull request by hwilmers**
This is a collection of small fixes to the ublox-sara-r4 modem driver.
Two of these fixes are contained in earlier (still unmerged) pull requests.
- PR24141: fix calculation of RSSI for modem Sara U201
- PR24130: fix closing of UDP socket
- Set ENOTSUP when sendmsg is called

Details are contained in the individual commit messages.

Signed-off-by: Hans Wilmers <hans@wilmers.no>

### CODE REVIEW

**Comment by hwilmers** [comment]
I have now closed the previous PR #24130 and #24141, as they are fully contained in this PR.

Sorry for any confusion I may have caused by this!

File: drivers/modem/ublox-sara-r4.c [hunk 1]
```diff
@@ -1252,6 +1252,12 @@ static ssize_t offload_write(void *obj, const void *buffer, size_t count)
 	return offload_sendto(obj, buffer, count, 0, NULL, 0);
 }
 
+static ssize_t offload_sendmsg(void *obj, const struct msghdr *msg, int flags)
+{
+	errno = -ENOTSUP;
```
Comment (jukkar) [inline drivers/modem/ublox-sara-r4.c:1257]: I think the errno value should be >0, just just do `errno = ENOTSUP;` here.

**Review by jukkar — CHANGES_REQUESTED** [review]
LGTM, minor issue with errno value.

File: drivers/modem/ublox-sara-r4.c (hunk 1 above)
Comment (hwilmers) [reply drivers/modem/ublox-sara-r4.c:1257]: Ah, of course - thanks for the note.

As a matter of fact I will make an attempt to implement sendmsg after all, following the discussion in #24519. So this code here is going to be replaced soon.

Comment (hwilmers) [reply drivers/modem/ublox-sara-r4.c]: I have now implemented sendmsg, and included in this PR.
This is tested for TCP using the MQTT library, but no tests are done for UDP packets.

Comment (hwilmers) [reply drivers/modem/ublox-sara-r4.c]: > I think the errno value should be >0, just just do `errno = ENOTSUP;` here.

Then there may be a similar problem in the simplelink wifi driver, where I borrowed these lines:
https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/wifi/simplelink/simplelink_sockets.c#L963

Comment (jukkar) [reply drivers/modem/ublox-sara-r4.c]: > Then there may be a similar problem in the simplelink wifi driver, where I borrowed these lines:
> https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/wifi/simplelink/simplelink_sockets.c#L966

Yep, that is a bug in simplelink driver.

**Review by jukkar — APPROVED** [review]
LGTM

**Review by mike-scott — APPROVED** [review]
LGTM. Thank you for submitting!

**Pull request merged**